Hmm... Windows failures look related (may be something in the test, or because `git version 2.11.1.windows.1`, which looks really old);

<details>

```
2022-10-18T23:01:44.5344656Z === Failed
2022-10-18T23:01:44.5345106Z === FAIL: github.com/docker/docker/builder/remotecontext/git TestCheckoutGit (unknown)
2022-10-18T23:01:44.5345719Z     gitutils_test.go:180: git version 2.11.1.windows.1
2022-10-18T23:01:44.5346451Z     gitutils_test.go:239: Initialized empty Git repository in C:/Users/ContainerAdministrator/AppData/Local/Temp/TestCheckoutGit2190731250/001/repo/.git/
2022-10-18T23:01:44.5347183Z     gitutils_test.go:253: warning: LF will be replaced by CRLF in subdir/Dockerfile.
2022-10-18T23:01:44.5347721Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5348788Z     gitutils_test.go:254: [master (root-commit) 9ce5de0] First commit
2022-10-18T23:01:44.5349234Z          2 files changed, 3 insertions(+)
2022-10-18T23:01:44.5349592Z          create mode 100644 Dockerfile
2022-10-18T23:01:44.5349966Z          create mode 100644 subdir/Dockerfile
2022-10-18T23:01:44.5350523Z         warning: LF will be replaced by CRLF in subdir/Dockerfile.
2022-10-18T23:01:44.5351038Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5351717Z     gitutils_test.go:255: Switched to a new branch 'test'
2022-10-18T23:01:44.5352255Z     gitutils_test.go:260: warning: LF will be replaced by CRLF in Dockerfile.
2022-10-18T23:01:44.5352851Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5353368Z         warning: LF will be replaced by CRLF in subdir/Dockerfile.
2022-10-18T23:01:44.5354020Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5354661Z     gitutils_test.go:261: warning: LF will be replaced by CRLF in Dockerfile.
2022-10-18T23:01:44.5355323Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5355797Z         [test c4de4ac] Branch commit
2022-10-18T23:01:44.5356286Z          2 files changed, 3 insertions(+), 2 deletions(-)
2022-10-18T23:01:44.5357013Z         warning: LF will be replaced by CRLF in subdir/Dockerfile.
2022-10-18T23:01:44.5357623Z         The file will have its original line endings in your working directory.
2022-10-18T23:01:44.5358202Z     gitutils_test.go:262: Switched to branch 'master'
2022-10-18T23:01:44.5358962Z     gitutils_test.go:266: Initialized empty Git repository in C:/Users/ContainerAdministrator/AppData/Local/Temp/TestCheckoutGit2190731250/001/subrepo/.git/
2022-10-18T23:01:44.5359960Z     gitutils_test.go:273: [master (root-commit) 7d25644] Subrepo initial
2022-10-18T23:01:44.5360425Z          1 file changed, 1 insertion(+)
2022-10-18T23:01:44.5360824Z          create mode 100644 subfile
2022-10-18T23:01:44.5361193Z coverage: 46.6% of statements
2022-10-18T23:01:44.5361579Z panic: test timed out after 10m0s
2022-10-18T23:01:44.5361888Z 
2022-10-18T23:01:44.5361979Z goroutine 43 [running]:
2022-10-18T23:01:44.5362384Z testing.(*M).startAlarm.func1()
2022-10-18T23:01:44.5362797Z 	c:/go/src/testing/testing.go:2036 +0x8e
2022-10-18T23:01:44.5363193Z created by time.goFunc
2022-10-18T23:01:44.5363583Z 	c:/go/src/time/sleep.go:176 +0x32
2022-10-18T23:01:44.5363879Z 
2022-10-18T23:01:44.5363971Z goroutine 1 [chan receive]:
2022-10-18T23:01:44.5364537Z testing.(*T).Run(0xc000043a00, {0xf835dd?, 0xb0fc13?}, 0xfbbe78)
2022-10-18T23:01:44.5364948Z 	c:/go/src/testing/testing.go:1494 +0x37a
2022-10-18T23:01:44.5365377Z testing.runTests.func1(0xc000146a20?)
2022-10-18T23:01:44.5365811Z 	c:/go/src/testing/testing.go:1846 +0x6e
2022-10-18T23:01:44.5366220Z testing.tRunner(0xc000043a00, 0xc000107ca0)
2022-10-18T23:01:44.5366637Z 	c:/go/src/testing/testing.go:1446 +0x10b
2022-10-18T23:01:44.5367234Z testing.runTests(0xc000000fa0?, {0x12da2c0, 0x7, 0x7}, {0x1dcd3500108?, 0x40?, 0x12e2ac0?})
2022-10-18T23:01:44.5367652Z 	c:/go/src/testing/testing.go:1844 +0x456
2022-10-18T23:01:44.5368038Z testing.(*M).Run(0xc000000fa0)
2022-10-18T23:01:44.5368404Z 	c:/go/src/testing/testing.go:1726 +0x5d9
2022-10-18T23:01:44.5368762Z main.main()
2022-10-18T23:01:44.5369096Z 	_testmain.go:107 +0x255
2022-10-18T23:01:44.5369369Z 
2022-10-18T23:01:44.5369490Z goroutine 35 [syscall, locked to thread]:
2022-10-18T23:01:44.5369895Z syscall.SyscallN(0x7ffb7f423920?, {0xc000069950?, 0x3?, 0x0?})
2022-10-18T23:01:44.5370498Z 	c:/go/src/runtime/syscall_windows.go:557 +0x109
2022-10-18T23:01:44.5370900Z syscall.Syscall(0x2030000?, 0xd?, 0x30000?, 0x12?, 0x0?)
2022-10-18T23:01:44.5371347Z 	c:/go/src/runtime/syscall_windows.go:495 +0x3b
2022-10-18T23:01:44.5371794Z syscall.WaitForSingleObject(0x1dcd356f840?, 0xffffffff)
2022-10-18T23:01:44.5372375Z 	c:/go/src/syscall/zsyscall_windows.go:1145 +0x65
2022-10-18T23:01:44.5372792Z os.(*Process).wait(0xc0001f46c0)
2022-10-18T23:01:44.5373195Z 	c:/go/src/os/exec_windows.go:18 +0x65
2022-10-18T23:01:44.5373584Z os.(*Process).Wait(...)
2022-10-18T23:01:44.5373957Z 	c:/go/src/os/exec.go:132
2022-10-18T23:01:44.5374286Z os/exec.(*Cmd).Wait(0xc000135e40)
2022-10-18T23:01:44.5374635Z 	c:/go/src/os/exec/exec.go:599 +0x4b
2022-10-18T23:01:44.5375028Z os/exec.(*Cmd).Run(0x6?)
2022-10-18T23:01:44.5375399Z 	c:/go/src/os/exec/exec.go:437 +0x39
2022-10-18T23:01:44.5375811Z os/exec.(*Cmd).CombinedOutput(0xc000135e40)
2022-10-18T23:01:44.5376164Z 	c:/go/src/os/exec/exec.go:707 +0xa8
2022-10-18T23:01:44.5376846Z github.com/docker/docker/builder/remotecontext/git.gitRepo.gitWithinDir({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x0}, {0xc0000c00a0, 0x47}, {0xc000069d60, ...})
2022-10-18T23:01:44.5377980Z 	C:/gopath/src/github.com/docker/docker/builder/remotecontext/git/gitutils.go:224 +0x35a
2022-10-18T23:01:44.5378660Z github.com/docker/docker/builder/remotecontext/git.TestCheckoutGit(0xc0001bd380)
2022-10-18T23:01:44.5379381Z 	C:/gopath/src/github.com/docker/docker/builder/remotecontext/git/gitutils_test.go:275 +0x1545
2022-10-18T23:01:44.5379888Z testing.tRunner(0xc0001bd380, 0xfbbe78)
2022-10-18T23:01:44.5380301Z 	c:/go/src/testing/testing.go:1446 +0x10b
2022-10-18T23:01:44.5380857Z created by testing.(*T).Run
2022-10-18T23:01:44.5381341Z 	c:/go/src/testing/testing.go:1493 +0x35f
2022-10-18T23:01:44.5381662Z 
2022-10-18T23:01:44.5381747Z goroutine 19 [IO wait]:
2022-10-18T23:01:44.5382337Z internal/poll.runtime_pollWait(0x1dcd35c3c40, 0x72)
2022-10-18T23:01:44.5382739Z 	c:/go/src/runtime/netpoll.go:305 +0x89
2022-10-18T23:01:44.5383190Z internal/poll.(*pollDesc).wait(0xc000053ba8?, 0x0?, 0x0)
2022-10-18T23:01:44.5383798Z 	c:/go/src/internal/poll/fd_poll_runtime.go:84 +0x32
2022-10-18T23:01:44.5384231Z internal/poll.execIO(0xc000005698, 0xc000053bf8)
2022-10-18T23:01:44.5384645Z 	c:/go/src/internal/poll/fd_windows.go:175 +0xe5
2022-10-18T23:01:44.5385304Z internal/poll.(*FD).acceptOne(0xc000005680, 0x174, {0xc0001a00f0?, 0xc00002dad0?, 0x0?}, 0x0?)
2022-10-18T23:01:44.5385779Z 	c:/go/src/internal/poll/fd_windows.go:942 +0x6d
2022-10-18T23:01:44.5386365Z internal/poll.(*FD).Accept(0xc000005680, 0xc000053dd0)
2022-10-18T23:01:44.5386835Z 	c:/go/src/internal/poll/fd_windows.go:976 +0x1d6
2022-10-18T23:01:44.5387273Z net.(*netFD).accept(0xc000005680)
2022-10-18T23:01:44.5387684Z 	c:/go/src/net/fd_windows.go:139 +0x65
2022-10-18T23:01:44.5388256Z net.(*TCPListener).accept(0xc0000084c8)
2022-10-18T23:01:44.5388659Z 	c:/go/src/net/tcpsock_posix.go:142 +0x28
2022-10-18T23:01:44.5389094Z net.(*TCPListener).Accept(0xc0000084c8)
2022-10-18T23:01:44.5389469Z 	c:/go/src/net/tcpsock.go:288 +0x3d
2022-10-18T23:01:44.5389910Z net/http.(*Server).Serve(0xc00019c000, {0x104b320, 0xc0000084c8})
2022-10-18T23:01:44.5390495Z 	c:/go/src/net/http/server.go:3070 +0x385
2022-10-18T23:01:44.5390945Z net/http/httptest.(*Server).goServe.func1()
2022-10-18T23:01:44.5391394Z 	c:/go/src/net/http/httptest/server.go:310 +0x6a
2022-10-18T23:01:44.5391981Z created by net/http/httptest.(*Server).goServe
2022-10-18T23:01:44.5392433Z 	c:/go/src/net/http/httptest/server.go:308 +0x6a
2022-10-18T23:01:44.5392762Z 
2022-10-18T23:01:44.5392849Z goroutine 27 [IO wait]:
2022-10-18T23:01:44.5393260Z internal/poll.runtime_pollWait(0x1dcd35c3970, 0x72)
2022-10-18T23:01:44.5393860Z 	c:/go/src/runtime/netpoll.go:305 +0x89
2022-10-18T23:01:44.5394338Z internal/poll.(*pollDesc).wait(0xc000059ba8?, 0x0?, 0x0)
2022-10-18T23:01:44.5394927Z 	c:/go/src/internal/poll/fd_poll_runtime.go:84 +0x32
2022-10-18T23:01:44.5395395Z internal/poll.execIO(0xc0001be018, 0xc000059bf8)
2022-10-18T23:01:44.5395841Z 	c:/go/src/internal/poll/fd_windows.go:175 +0xe5
2022-10-18T23:01:44.5396479Z internal/poll.(*FD).acceptOne(0xc0001be000, 0x184, {0xc0001a02d0?, 0xc00002dd40?, 0x0?}, 0x0?)
2022-10-18T23:01:44.5396971Z 	c:/go/src/internal/poll/fd_windows.go:942 +0x6d
2022-10-18T23:01:44.5397598Z internal/poll.(*FD).Accept(0xc0001be000, 0xc000059dd0)
2022-10-18T23:01:44.5398066Z 	c:/go/src/internal/poll/fd_windows.go:976 +0x1d6
2022-10-18T23:01:44.5398499Z net.(*netFD).accept(0xc0001be000)
2022-10-18T23:01:44.5399056Z 	c:/go/src/net/fd_windows.go:139 +0x65
2022-10-18T23:01:44.5399441Z net.(*TCPListener).accept(0xc0000085d0)
2022-10-18T23:01:44.5399883Z 	c:/go/src/net/tcpsock_posix.go:142 +0x28
2022-10-18T23:01:44.5400306Z net.(*TCPListener).Accept(0xc0000085d0)
2022-10-18T23:01:44.5400880Z 	c:/go/src/net/tcpsock.go:288 +0x3d
2022-10-18T23:01:44.5401327Z net/http.(*Server).Serve(0xc00019c0f0, {0x104b320, 0xc0000085d0})
2022-10-18T23:01:44.5401759Z 	c:/go/src/net/http/server.go:3070 +0x385
2022-10-18T23:01:44.5402192Z net/http/httptest.(*Server).goServe.func1()
2022-10-18T23:01:44.5402760Z 	c:/go/src/net/http/httptest/server.go:310 +0x6a
2022-10-18T23:01:44.5403485Z created by net/http/httptest.(*Server).goServe
2022-10-18T23:01:44.5403937Z 	c:/go/src/net/http/httptest/server.go:308 +0x6a
2022-10-18T23:01:44.5404268Z 
2022-10-18T23:01:44.5404341Z goroutine 50 [IO wait]:
2022-10-18T23:01:44.5404770Z internal/poll.runtime_pollWait(0x1dcd35c36a0, 0x72)
2022-10-18T23:01:44.5405348Z 	c:/go/src/runtime/netpoll.go:305 +0x89
2022-10-18T23:01:44.5405817Z internal/poll.(*pollDesc).wait(0x1dcf91620f8?, 0x3?, 0x0)
2022-10-18T23:01:44.5406410Z 	c:/go/src/internal/poll/fd_poll_runtime.go:84 +0x32
2022-10-18T23:01:44.5406965Z internal/poll.execIO(0xc0000a4018, 0xc0001b5bf8)
2022-10-18T23:01:44.5407353Z 	c:/go/src/internal/poll/fd_windows.go:175 +0xe5
2022-10-18T23:01:44.5408022Z internal/poll.(*FD).acceptOne(0xc0000a4000, 0x1cc, {0xc0001a0870?, 0x1dcf91620f8?, 0xd0?}, 0x1dcd3500108?)
2022-10-18T23:01:44.5408692Z 	c:/go/src/internal/poll/fd_windows.go:942 +0x6d
2022-10-18T23:01:44.5409160Z internal/poll.(*FD).Accept(0xc0000a4000, 0xc0001b5dd0)
2022-10-18T23:01:44.5409756Z 	c:/go/src/internal/poll/fd_windows.go:976 +0x1d6
2022-10-18T23:01:44.5410178Z net.(*netFD).accept(0xc0000a4000)
2022-10-18T23:01:44.5410568Z 	c:/go/src/net/fd_windows.go:139 +0x65
2022-10-18T23:01:44.5410990Z net.(*TCPListener).accept(0xc000092060)
2022-10-18T23:01:44.5411438Z 	c:/go/src/net/tcpsock_posix.go:142 +0x28
2022-10-18T23:01:44.5412005Z net.(*TCPListener).Accept(0xc000092060)
2022-10-18T23:01:44.5412419Z 	c:/go/src/net/tcpsock.go:288 +0x3d
2022-10-18T23:01:44.5412869Z net/http.(*Server).Serve(0xc0000a8000, {0x104b320, 0xc000092060})
2022-10-18T23:01:44.5413492Z 	c:/go/src/net/http/server.go:3070 +0x385
2022-10-18T23:01:44.5413954Z net/http/httptest.(*Server).goServe.func1()
2022-10-18T23:01:44.5414412Z 	c:/go/src/net/http/httptest/server.go:310 +0x6a
2022-10-18T23:01:44.5414971Z created by net/http/httptest.(*Server).goServe
2022-10-18T23:01:44.5415414Z 	c:/go/src/net/http/httptest/server.go:308 +0x6a
2022-10-18T23:01:44.5415705Z 
2022-10-18T23:01:44.5415822Z goroutine 42 [syscall, locked to thread]:
2022-10-18T23:01:44.5416453Z syscall.SyscallN(0x0?, {0xc0001c9c70?, 0xc0001c9ca0?, 0xac3305?})
2022-10-18T23:01:44.5416906Z 	c:/go/src/runtime/syscall_windows.go:557 +0x109
2022-10-18T23:01:44.5417520Z syscall.Syscall6(0xc0000841a0?, 0xc0001c9d30?, 0xaae3df?, 0xae5385?, 0xc0000841a0?, 0xc0001c9d18?, 0xb07a07?, 0x100c0000841a0?)
2022-10-18T23:01:44.5417984Z 	c:/go/src/runtime/syscall_windows.go:501 +0x50
2022-10-18T23:01:44.5418549Z syscall.readFile(0xb0ec80?, {0xc000086c00?, 0x200, 0x800000?}, 0x7ffff800000?, 0x2?)
2022-10-18T23:01:44.5419042Z 	c:/go/src/syscall/zsyscall_windows.go:1024 +0x94
2022-10-18T23:01:44.5419411Z syscall.ReadFile(...)
2022-10-18T23:01:44.5419956Z 	c:/go/src/syscall/syscall_windows.go:400
2022-10-18T23:01:44.5420411Z syscall.Read(0xc0001f8280?, {0xc000086c00?, 0xbab3be?, 0xc000086c00?})
2022-10-18T23:01:44.5420864Z 	c:/go/src/syscall/syscall_windows.go:379 +0x2f
2022-10-18T23:01:44.5421469Z internal/poll.(*FD).Read(0xc0001f8280, {0xc000086c00, 0x200, 0x200})
2022-10-18T23:01:44.5421903Z 	c:/go/src/internal/poll/fd_windows.go:427 +0x1b1
2022-10-18T23:01:44.5422318Z os.(*File).read(...)
2022-10-18T23:01:44.5422838Z 	c:/go/src/os/file_posix.go:31
2022-10-18T23:01:44.5423296Z os.(*File).Read(0xc00000a200, {0xc000086c00?, 0x0?, 0xc0001c9ea0?})
2022-10-18T23:01:44.5423714Z 	c:/go/src/os/file.go:118 +0x5e
2022-10-18T23:01:44.5424168Z bytes.(*Buffer).ReadFrom(0xc0001f0810, {0x1047aa0, 0xc00000a200})
2022-10-18T23:01:44.5424771Z 	c:/go/src/bytes/buffer.go:202 +0x98
2022-10-18T23:01:44.5425214Z io.copyBuffer({0x10470e0, 0xc0001f0810}, {0x1047aa0, 0xc00000a200}, {0x0, 0x0, 0x0})
2022-10-18T23:01:44.5425829Z 	c:/go/src/io/io.go:413 +0x14b
2022-10-18T23:01:44.5426194Z io.Copy(...)
2022-10-18T23:01:44.5426564Z 	c:/go/src/io/io.go:386
2022-10-18T23:01:44.5426982Z os/exec.(*Cmd).writerDescriptor.func1()
2022-10-18T23:01:44.5427427Z 	c:/go/src/os/exec/exec.go:407 +0x3a
2022-10-18T23:01:44.5428131Z os/exec.(*Cmd).Start.func1(0x0?)
2022-10-18T23:01:44.5428531Z 	c:/go/src/os/exec/exec.go:544 +0x25
2022-10-18T23:01:44.5428941Z created by os/exec.(*Cmd).Start
2022-10-18T23:01:44.5429322Z 	c:/go/src/os/exec/exec.go:543 +0x79e
2022-10-18T23:01:44.5429635Z 
2022-10-18T23:01:44.5429990Z === FAIL: github.com/docker/docker/pkg/archive TestIsArchivePathDir (15.04s)
2022-10-18T23:01:44.5431218Z     archive_test.go:59: Fail to create an archive file for test :       0 [main] sh 3220 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5431917Z         Stack trace:
2022-10-18T23:01:44.5432605Z         Frame        Function    Args
2022-10-18T23:01:44.5433087Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5433721Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E3450D97D9FE, 00000000000)
2022-10-18T23:01:44.5434245Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5434894Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E3451689AADA)
2022-10-18T23:01:44.5435554Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5435981Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5436394Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5437005Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5437430Z         End of stack trace
2022-10-18T23:01:44.5437802Z         .
2022-10-18T23:01:44.5437917Z 
2022-10-18T23:01:44.5438290Z === FAIL: github.com/docker/docker/pkg/archive TestIsArchivePathInvalidFile (15.04s)
2022-10-18T23:01:44.5439446Z     archive_test.go:70: Fail to create an archive file for test :       4 [main] sh 4928 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5440090Z         Stack trace:
2022-10-18T23:01:44.5440447Z         Frame        Function    Args
2022-10-18T23:01:44.5440858Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5441515Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E345168F3064, 00000000000)
2022-10-18T23:01:44.5441971Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5442642Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E3451F806AD0)
2022-10-18T23:01:44.5443252Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5443733Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5444134Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5444562Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5445139Z         End of stack trace
2022-10-18T23:01:44.5445491Z         .
2022-10-18T23:01:44.5445596Z 
2022-10-18T23:01:44.5445960Z === FAIL: github.com/docker/docker/pkg/archive TestIsArchivePathTar (15.04s)
2022-10-18T23:01:44.5446992Z     archive_test.go:86: Fail to create an archive file for test :       0 [main] sh 1368 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5447675Z         Stack trace:
2022-10-18T23:01:44.5448078Z         Frame        Function    Args
2022-10-18T23:01:44.5448527Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5449132Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E3451F85E10A, 00000000000)
2022-10-18T23:01:44.5449773Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5450502Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E345287792A5)
2022-10-18T23:01:44.5450983Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5451412Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5451915Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5452389Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5452777Z         End of stack trace
2022-10-18T23:01:44.5453198Z         .
2022-10-18T23:01:44.5453456Z 
2022-10-18T23:01:44.5453783Z === FAIL: github.com/docker/docker/pkg/archive TestDecompressStreamGzip (15.04s)
2022-10-18T23:01:44.5454911Z     archive_test.go:101: Failed to create an archive file for test :       0 [main] sh 2772 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5455555Z         Stack trace:
2022-10-18T23:01:44.5455922Z         Frame        Function    Args
2022-10-18T23:01:44.5456319Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5456819Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E345287D2158, 00000000000)
2022-10-18T23:01:44.5457480Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5458155Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E345316ECD0B)
2022-10-18T23:01:44.5458605Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5459462Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5473090Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5473388Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5473628Z         End of stack trace
2022-10-18T23:01:44.5473981Z         .
2022-10-18T23:01:44.5474241Z 
2022-10-18T23:01:44.5474575Z === FAIL: github.com/docker/docker/pkg/archive TestDecompressStreamBzip2 (15.04s)
2022-10-18T23:01:44.5475576Z     archive_test.go:101: Failed to create an archive file for test :       0 [main] sh 4804 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5476238Z         Stack trace:
2022-10-18T23:01:44.5476610Z         Frame        Function    Args
2022-10-18T23:01:44.5477157Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5477662Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E3453172F05E, 00000000000)
2022-10-18T23:01:44.5477994Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5478293Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E3453A65C6C2)
2022-10-18T23:01:44.5478733Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5479630Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5480027Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5481135Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5481521Z         End of stack trace
2022-10-18T23:01:44.5482166Z         .
2022-10-18T23:01:44.5482306Z 
2022-10-18T23:01:44.5482621Z === FAIL: github.com/docker/docker/pkg/archive TestCmdStreamLargeStderr (5.01s)
2022-10-18T23:01:44.5483324Z     archive_test.go:246: Command did not complete in 5 seconds; probable deadlock
2022-10-18T23:01:44.5483935Z 
2022-10-18T23:01:44.5484189Z === FAIL: github.com/docker/docker/pkg/archive TestCmdStreamGood (15.04s)
2022-10-18T23:01:44.5485535Z     archive_test.go:276: Command should not have failed (err=exit status 256:       0 [main] sh 5520 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5486045Z         Stack trace:
2022-10-18T23:01:44.5486732Z         Frame        Function    Args
2022-10-18T23:01:44.5487119Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5487849Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E3453D6AE426, 00000000000)
2022-10-18T23:01:44.5488577Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5489061Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E345465D3A47)
2022-10-18T23:01:44.5489825Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5490368Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5490766Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5491176Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5491540Z         End of stack trace
2022-10-18T23:01:44.5491918Z         )
2022-10-18T23:01:44.5492158Z 
2022-10-18T23:01:44.5492537Z === FAIL: github.com/docker/docker/pkg/archive TestUntarPathWithInvalidDest (15.06s)
2022-10-18T23:01:44.5493052Z     archive_test.go:303: assertion failed: error is not nil: exit status 256
2022-10-18T23:01:44.5493386Z 
2022-10-18T23:01:44.5493668Z === FAIL: github.com/docker/docker/pkg/archive TestUntarPath (15.05s)
2022-10-18T23:01:44.5494475Z     archive_test.go:347: assertion failed: error is not nil: exit status 256
2022-10-18T23:01:44.5495111Z 
2022-10-18T23:01:44.5495454Z === FAIL: github.com/docker/docker/pkg/archive TestUntarPathWithDestinationFile (15.04s)
2022-10-18T23:01:44.5495931Z     archive_test.go:381: exit status 256
2022-10-18T23:01:44.5496225Z 
2022-10-18T23:01:44.5496672Z === FAIL: github.com/docker/docker/pkg/archive TestUntarPathWithDestinationSrcFileAsFolder (15.03s)
2022-10-18T23:01:44.5497282Z     archive_test.go:418: exit status 256
2022-10-18T23:01:44.5497603Z 
2022-10-18T23:01:44.5497907Z === FAIL: github.com/docker/docker/pkg/archive TestDisablePigz (15.09s)
2022-10-18T23:01:44.5498385Z     archive_test.go:1420: Test will not check full path when Pigz not installed
2022-10-18T23:01:44.5499435Z     archive_test.go:101: Failed to create an archive file for test :       0 [main] sh 5568 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5500055Z         Stack trace:
2022-10-18T23:01:44.5500454Z         Frame        Function    Args
2022-10-18T23:01:44.5501251Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5501710Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E345703BD8BC, 00000000000)
2022-10-18T23:01:44.5502278Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5502744Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E345792EC879)
2022-10-18T23:01:44.5503512Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5504609Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5505846Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5506327Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5506703Z         End of stack trace
2022-10-18T23:01:44.5507063Z         .
2022-10-18T23:01:44.5507176Z 
2022-10-18T23:01:44.5507784Z === FAIL: github.com/docker/docker/pkg/archive TestPigz (15.04s)
2022-10-18T23:01:44.5509011Z     archive_test.go:101: Failed to create an archive file for test :       0 [main] sh 2208 C:\git\usr\bin\sh.exe: *** fatal error - add_item ("\??\C:\git", "/", ...) failed, errno 1
2022-10-18T23:01:44.5510015Z         Stack trace:
2022-10-18T23:01:44.5510377Z         Frame        Function    Args
2022-10-18T23:01:44.5510843Z         000FFFF9BB0  0018005CCCE (00180241542, 0018021EC39, 000FFFF9BB0, 000FFFF8B30)
2022-10-18T23:01:44.5511319Z         000FFFF9BB0  00180046419 (000FFFFABC4, 000FFFF9BB0, 1D8E34579338B73, 00000000000)
2022-10-18T23:01:44.5511916Z         000FFFF9BB0  00180046452 (000FFFF9BB0, 00000000001, 000FFFF9BB0, 675C3A435C3F3F5C)
2022-10-18T23:01:44.5512628Z         000FFFF9BB0  001800CB8AB (00000000000, 000FFFFCE00, 001800BB2D8, 1D8E3458225E3F8)
2022-10-18T23:01:44.5513144Z         000FFFFCC00  001801195B5 (00000000000, 00000000000, 001800B39DE, 00000000000)
2022-10-18T23:01:44.5514071Z         000FFFFCCC0  00180046A45 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5514592Z         00000000000  001800456B3 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5515007Z         000FFFFFFF0  00180045764 (00000000000, 00000000000, 00000000000, 00000000000)
2022-10-18T23:01:44.5515404Z         End of stack trace
2022-10-18T23:01:44.5515781Z         .
2022-10-18T23:01:44.5515895Z 
2022-10-18T23:01:44.5516173Z DONE 2537 tests, 70 skipped, 14 failures in 798.739s
2022-10-18T23:01:44.5516471Z 
2022-10-18T23:01:44.5516565Z ERROR: make.ps1 failed:
```

<details>

--- a/builder/remotecontext/git.go
+++ b/builder/remotecontext/git.go
@@ -11,7 +11,7 @@ import (
 
 // MakeGitContext returns a Context from gitURL that is cloned in a temporary directory.
 func MakeGitContext(gitURL string) (builder.Source, error) {
-	root, err := git.Clone(gitURL)
+	root, err := git.Clone(gitURL, git.WithIsolatedConfig(true))
 	if err != nil {
 		return nil, err
 	}

--- a/builder/remotecontext/git/gitutils.go
+++ b/builder/remotecontext/git/gitutils.go
@@ -16,21 +16,37 @@ type gitRepo struct {
 	remote string
 	ref    string
 	subdir string
+
+	isolateConfig bool
+}
+
+type CloneOption func(*gitRepo)
+
+// WithIsolatedConfig disables reading the user or system gitconfig files when
+// performing Git operations.
+func WithIsolatedConfig(v bool) CloneOption {
+	return func(gr *gitRepo) {
+		gr.isolateConfig = v
+	}
 }
 
 // Clone clones a repository into a newly created directory which
 // will be under "docker-build-git"
-func Clone(remoteURL string) (string, error) {
+func Clone(remoteURL string, opts ...CloneOption) (string, error) {
 	repo, err := parseRemoteURL(remoteURL)
 
 	if err != nil {
 		return "", err
 	}
 
-	return cloneGitRepo(repo)
+	for _, opt := range opts {
+		opt(&repo)
+	}
+
+	return repo.clone()
 }
 
-func cloneGitRepo(repo gitRepo) (checkoutDir string, err error) {
+func (repo gitRepo) clone() (checkoutDir string, err error) {
 	fetch := fetchArgs(repo.remote, repo.ref)
 
 	root, err := os.MkdirTemp("", "docker-build-git")
@@ -44,21 +60,21 @@ func cloneGitRepo(repo gitRepo) (checkoutDir string, err error) {
 		}
 	}()
 
-	if out, err := gitWithinDir(root, "init"); err != nil {
+	if out, err := repo.gitWithinDir(root, "init"); err != nil {
 		return "", errors.Wrapf(err, "failed to init repo at %s: %s", root, out)
 	}
 
 	// Add origin remote for compatibility with previous implementation that
 	// used "git clone" and also to make sure local refs are created for branches
-	if out, err := gitWithinDir(root, "remote", "add", "origin", repo.remote); err != nil {
+	if out, err := repo.gitWithinDir(root, "remote", "add", "origin", repo.remote); err != nil {
 		return "", errors.Wrapf(err, "failed add origin repo at %s: %s", repo.remote, out)
 	}
 
-	if output, err := gitWithinDir(root, fetch...); err != nil {
+	if output, err := repo.gitWithinDir(root, fetch...); err != nil {
 		return "", errors.Wrapf(err, "error fetching: %s", output)
 	}
 
-	checkoutDir, err = checkoutGit(root, repo.ref, repo.subdir)
+	checkoutDir, err = repo.checkout(root)
 	if err != nil {
 		return "", err
 	}
@@ -162,20 +178,20 @@ func supportsShallowClone(remoteURL string) bool {
 	return true
 }
 
-func checkoutGit(root, ref, subdir string) (string, error) {
+func (repo gitRepo) checkout(root string) (string, error) {
 	// Try checking out by ref name first. This will work on branches and sets
 	// .git/HEAD to the current branch name
-	if output, err := gitWithinDir(root, "checkout", ref); err != nil {
+	if output, err := repo.gitWithinDir(root, "checkout", repo.ref); err != nil {
 		// If checking out by branch name fails check out the last fetched ref
-		if _, err2 := gitWithinDir(root, "checkout", "FETCH_HEAD"); err2 != nil {
-			return "", errors.Wrapf(err, "error checking out %s: %s", ref, output)
+		if _, err2 := repo.gitWithinDir(root, "checkout", "FETCH_HEAD"); err2 != nil {
+			return "", errors.Wrapf(err, "error checking out %s: %s", repo.ref, output)
 		}
 	}
 
-	if subdir != "" {
-		newCtx, err := symlink.FollowSymlinkInScope(filepath.Join(root, subdir), root)
+	if repo.subdir != "" {
+		newCtx, err := symlink.FollowSymlinkInScope(filepath.Join(root, repo.subdir), root)
 		if err != nil {
-			return "", errors.Wrapf(err, "error setting git context, %q not within git root", subdir)
+			return "", errors.Wrapf(err, "error setting git context, %q not within git root", repo.subdir)
 		}
 
 		fi, err := os.Stat(newCtx)
@@ -191,15 +207,20 @@ func checkoutGit(root, ref, subdir string) (string, error) {
 	return root, nil
 }
 
-func gitWithinDir(dir string, args ...string) ([]byte, error) {
+func (repo gitRepo) gitWithinDir(dir string, args ...string) ([]byte, error) {
 	args = append([]string{"-c", "protocol.file.allow=never"}, args...) // Block sneaky repositories from using repos from the filesystem as submodules.
 	cmd := exec.Command("git", args...)
 	cmd.Dir = dir
-	cmd.Env = append(cmd.Env,
-		"GIT_PROTOCOL_FROM_USER=0", // Disable unsafe remote protocols.
-		"GIT_CONFIG_NOSYSTEM=1",    // Disable reading from system gitconfig.
-		"HOME=/dev/null",           // Disable reading from user gitconfig.
-	)
+	// Disable unsafe remote protocols.
+	cmd.Env = append(cmd.Env, "GIT_PROTOCOL_FROM_USER=0")
+
+	if repo.isolateConfig {
+		cmd.Env = append(cmd.Env,
+			"GIT_CONFIG_NOSYSTEM=1", // Disable reading from system gitconfig.
+			"HOME=/dev/null",        // Disable reading from user gitconfig.
+		)
+	}
+
 	return cmd.CombinedOutput()
 }
 

--- a/builder/remotecontext/git/gitutils.go
+++ b/builder/remotecontext/git/gitutils.go
@@ -20,6 +20,7 @@ type gitRepo struct {
 	isolateConfig bool
 }
 
+// CloneOption changes the behaviour of Clone().
 type CloneOption func(*gitRepo)
 
 // WithIsolatedConfig disables reading the user or system gitconfig files when

--- a/builder/remotecontext/git/gitutils.go
+++ b/builder/remotecontext/git/gitutils.go
@@ -192,12 +192,9 @@ func checkoutGit(root, ref, subdir string) (string, error) {
 }
 
 func gitWithinDir(dir string, args ...string) ([]byte, error) {
-	a := []string{"--work-tree", dir, "--git-dir", filepath.Join(dir, ".git")}
-	return git(append(a, args...)...)
-}
-
-func git(args ...string) ([]byte, error) {
-	return exec.Command("git", args...).CombinedOutput()
+	cmd := exec.Command("git", args...)
+	cmd.Dir = dir
+	return cmd.CombinedOutput()
 }
 
 // isGitTransport returns true if the provided str is a git transport by inspecting

--- a/builder/remotecontext/git/gitutils_test.go
+++ b/builder/remotecontext/git/gitutils_test.go
@@ -6,7 +6,6 @@ import (
 	"net/http/httptest"
 	"net/url"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"runtime"
 	"strings"
@@ -160,7 +159,7 @@ func TestCloneArgsGit(t *testing.T) {
 }
 
 func gitGetConfig(name string) string {
-	b, err := git([]string{"config", "--get", name}...)
+	b, err := gitWithinDir("", "config", "--get", name)
 	if err != nil {
 		// since we are interested in empty or non empty string,
 		// we can safely ignore the err here.
@@ -191,7 +190,7 @@ func TestCheckoutGit(t *testing.T) {
 	}
 
 	gitDir := filepath.Join(root, "repo")
-	must(git("-c", "init.defaultBranch=master", "init", gitDir))
+	must(gitWithinDir(root, "-c", "init.defaultBranch=master", "init", gitDir))
 	must(gitWithinDir(gitDir, "config", "user.email", "test@docker.com"))
 	must(gitWithinDir(gitDir, "config", "user.name", "Docker test"))
 	assert.NilError(t, os.WriteFile(filepath.Join(gitDir, "Dockerfile"), []byte("FROM scratch"), 0644))
@@ -218,7 +217,7 @@ func TestCheckoutGit(t *testing.T) {
 
 	// set up submodule
 	subrepoDir := filepath.Join(root, "subrepo")
-	must(git("-c", "init.defaultBranch=master", "init", subrepoDir))
+	must(gitWithinDir(root, "-c", "init.defaultBranch=master", "init", subrepoDir))
 	must(gitWithinDir(subrepoDir, "config", "user.email", "test@docker.com"))
 	must(gitWithinDir(subrepoDir, "config", "user.name", "Docker test"))
 
@@ -227,10 +226,7 @@ func TestCheckoutGit(t *testing.T) {
 	must(gitWithinDir(subrepoDir, "add", "-A"))
 	must(gitWithinDir(subrepoDir, "commit", "-am", "Subrepo initial"))
 
-	cmd := exec.Command("git", "submodule", "add", subrepoDir, "sub") // this command doesn't work with --work-tree
-	cmd.Dir = gitDir
-	must(cmd.CombinedOutput())
-
+	must(gitWithinDir(gitDir, "submodule", "add", subrepoDir, "sub"))
 	must(gitWithinDir(gitDir, "add", "-A"))
 	must(gitWithinDir(gitDir, "commit", "-am", "With submodule"))
 

--- a/builder/remotecontext/git/gitutils_test.go
+++ b/builder/remotecontext/git/gitutils_test.go
@@ -162,7 +162,7 @@ func TestCloneArgsGit(t *testing.T) {
 }
 
 func gitGetConfig(name string) string {
-	b, err := gitWithinDir("", "config", "--get", name)
+	b, err := gitRepo{}.gitWithinDir("", "config", "--get", name)
 	if err != nil {
 		// since we are interested in empty or non empty string,
 		// we can safely ignore the err here.
@@ -236,9 +236,9 @@ func TestCheckoutGit(t *testing.T) {
 	}
 
 	gitDir := filepath.Join(root, "repo")
-	must(gitWithinDir(root, "-c", "init.defaultBranch=master", "init", gitDir))
-	must(gitWithinDir(gitDir, "config", "user.email", "test@docker.com"))
-	must(gitWithinDir(gitDir, "config", "user.name", "Docker test"))
+	must(gitRepo{}.gitWithinDir(root, "-c", "init.defaultBranch=master", "init", gitDir))
+	must(gitRepo{}.gitWithinDir(gitDir, "config", "user.email", "test@docker.com"))
+	must(gitRepo{}.gitWithinDir(gitDir, "config", "user.name", "Docker test"))
 	assert.NilError(t, os.WriteFile(filepath.Join(gitDir, "Dockerfile"), []byte("FROM scratch"), 0644))
 
 	subDir := filepath.Join(gitDir, "subdir")
@@ -250,31 +250,31 @@ func TestCheckoutGit(t *testing.T) {
 		assert.NilError(t, os.Symlink("/subdir", filepath.Join(gitDir, "absolutelink")))
 	}
 
-	must(gitWithinDir(gitDir, "add", "-A"))
-	must(gitWithinDir(gitDir, "commit", "-am", "First commit"))
-	must(gitWithinDir(gitDir, "checkout", "-b", "test"))
+	must(gitRepo{}.gitWithinDir(gitDir, "add", "-A"))
+	must(gitRepo{}.gitWithinDir(gitDir, "commit", "-am", "First commit"))
+	must(gitRepo{}.gitWithinDir(gitDir, "checkout", "-b", "test"))
 
 	assert.NilError(t, os.WriteFile(filepath.Join(gitDir, "Dockerfile"), []byte("FROM scratch\nEXPOSE 3000"), 0644))
 	assert.NilError(t, os.WriteFile(filepath.Join(subDir, "Dockerfile"), []byte("FROM busybox\nEXPOSE 5000"), 0644))
 
-	must(gitWithinDir(gitDir, "add", "-A"))
-	must(gitWithinDir(gitDir, "commit", "-am", "Branch commit"))
-	must(gitWithinDir(gitDir, "checkout", "master"))
+	must(gitRepo{}.gitWithinDir(gitDir, "add", "-A"))
+	must(gitRepo{}.gitWithinDir(gitDir, "commit", "-am", "Branch commit"))
+	must(gitRepo{}.gitWithinDir(gitDir, "checkout", "master"))
 
 	// set up submodule
 	subrepoDir := filepath.Join(root, "subrepo")
-	must(gitWithinDir(root, "-c", "init.defaultBranch=master", "init", subrepoDir))
-	must(gitWithinDir(subrepoDir, "config", "user.email", "test@docker.com"))
-	must(gitWithinDir(subrepoDir, "config", "user.name", "Docker test"))
+	must(gitRepo{}.gitWithinDir(root, "-c", "init.defaultBranch=master", "init", subrepoDir))
+	must(gitRepo{}.gitWithinDir(subrepoDir, "config", "user.email", "test@docker.com"))
+	must(gitRepo{}.gitWithinDir(subrepoDir, "config", "user.name", "Docker test"))
 
 	assert.NilError(t, os.WriteFile(filepath.Join(subrepoDir, "subfile"), []byte("subcontents"), 0644))
 
-	must(gitWithinDir(subrepoDir, "add", "-A"))
-	must(gitWithinDir(subrepoDir, "commit", "-am", "Subrepo initial"))
+	must(gitRepo{}.gitWithinDir(subrepoDir, "add", "-A"))
+	must(gitRepo{}.gitWithinDir(subrepoDir, "commit", "-am", "Subrepo initial"))
 
-	must(gitWithinDir(gitDir, "submodule", "add", server.URL+"/subrepo", "sub"))
-	must(gitWithinDir(gitDir, "add", "-A"))
-	must(gitWithinDir(gitDir, "commit", "-am", "With submodule"))
+	must(gitRepo{}.gitWithinDir(gitDir, "submodule", "add", server.URL+"/subrepo", "sub"))
+	must(gitRepo{}.gitWithinDir(gitDir, "add", "-A"))
+	must(gitRepo{}.gitWithinDir(gitDir, "commit", "-am", "With submodule"))
 
 	type singleCase struct {
 		frag      string
@@ -311,7 +311,7 @@ func TestCheckoutGit(t *testing.T) {
 		t.Run(c.frag, func(t *testing.T) {
 			currentSubtest = t
 			ref, subdir := getRefAndSubdir(c.frag)
-			r, err := cloneGitRepo(gitRepo{remote: server.URL + "/repo", ref: ref, subdir: subdir})
+			r, err := gitRepo{remote: server.URL + "/repo", ref: ref, subdir: subdir}.clone()
 
 			if c.fail {
 				assert.Check(t, is.ErrorContains(err, ""))

